AppCenter/AppCenter/MSAppCenter.m

```diff
@@ -411,12 +407,14 @@ - (BOOL)startService:(Class)clazz
   }
 }
 
+- (NSString *)logUrl {
+  return _logUrl;
+}
+
 - (void)setLogUrl:(NSString *)logUrl {
-  @synchronized(self) {
-    _logUrl = logUrl;
-    if (self.channelGroup) {
-      [self.channelGroup setLogUrl:logUrl];
-    }
+  _logUrl = logUrl;
+  if (self.channelGroup) {
```

Comment:
This probably needs to be stored in a local variable and then tested + set since it isn't in a synchronized block

Reply:
@ElektrojungeMS @achocron Do we really need this null-check?
From the apple [docs](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/ProgrammingWithObjectiveC/WorkingwithObjects/WorkingwithObjects.html#//apple_ref/doc/uid/TP40011210-CH4-SW22)
> it’s perfectly acceptable in Objective-C to send a message to nil. If you do send a message to nil, obviously nothing happens.

Reply:
Thank you. I tested this method with `channelGroup = nil` (without `if` block) and can confirm that it doesn't cause issues.

In addition I verified calling this method from the test app — looks fine.

Reply:
Makes sense. Then I think the check is not needed.